src/main/java/io/jenkins/plugins/audit/listeners/ItemChangeListener.java

```diff
@@ -0,0 +1,137 @@
+package io.jenkins.plugins.audit.listeners;
+
+import hudson.Extension;
+import hudson.ExtensionList;
+import hudson.model.Item;
+import hudson.model.User;
+import io.jenkins.plugins.audit.event.*;
+import org.apache.logging.log4j.audit.LogEventFactory;
+
+import javax.annotation.Nonnull;
+import java.util.Date;
+
+/**
+ * Listener notified of any CRUD operations on items.
+ */
+@Extension
+public class ItemChangeListener extends hudson.model.listeners.ItemListener{
+
+    /**
+     * Fired when a new job is created and added to Jenkins, before the initial configuration page is provided.
+     *
+     * @param item is a job that is created
+     */
+    @Override
+    public void onCreated(@Nonnull Item item) {
+        CreateItem itemCreateEvent = LogEventFactory.getEvent(CreateItem.class);
+
+        itemCreateEvent.setItemName(item.getName());
+        itemCreateEvent.setItemUri(item.getUrl());
+        itemCreateEvent.setTimestamp(new Date().toString());
+        User user = User.current();
+        if(user != null)
+            itemCreateEvent.setUserId(user.getId());
+        else
+            itemCreateEvent.setUserId(null);
+
+        itemCreateEvent.logEvent();
+
+    }
+
+    /**
+     * Fired when a new job is created by copying from an existing job
+     *
+     * @param src is the source item that the new one was copied from.
+     * @param item is the newly created item.
+     */
+    @Override
+    public void onCopied(Item src, Item item) {
+        CopyItem itemCopyEvent = LogEventFactory.getEvent(CopyItem.class);
+
+        itemCopyEvent.setItemName(item.getName());
+        itemCopyEvent.setItemUri(item.getUrl());
```

Comment:
Similar to the node change event, we may wish to add an attribute to include both the source URI and destination URI. This way we can gather at a glance from the audit log event enough information to know what happened.

Reply:
Thanks, I added an attribute for that.